felix/fv/ipip_test.go

```diff
@@ -422,31 +398,52 @@ var _ = infrastructure.DatastoreDescribe("_BPF-SAFE_ IPIP topology before adding
 						Expect(err).NotTo(HaveOccurred())
 					}
 				}
-				c.Spec.ExternalNodesCIDRList = &[]string{addr, "1.1.1.1"}
+				c.Spec.ExternalNodesCIDRList = &[]string{addr}
 				log.WithFields(log.Fields{"felixconfiguration": c, "adding Addr": addr}).Info("Updating FelixConfiguration ")
 				_, err = client.FelixConfigurations().Update(ctx, c, options.SetOptions{})
 				Expect(err).NotTo(HaveOccurred())
 			}
-			updateConfig(prevBGPSpec.IPv4Address)
+
+			updateConfig(externalClient.IP)
 
 			// Wait for the config to take
 			for _, f := range felixes {
-				if bpfEnabled {
-					Eventually(f.BPFRoutes, "5s", "200ms").Should(ContainSubstring("1.1.1.1/32"))
-				} else {
-					Eventually(func() int {
-						return getNumIPSetMembers(f.Container, "cali40all-hosts-net")
-					}, "5s", "200ms").Should(Equal(3))
+				Eventually(func() int {
+					return getNumIPSetMembers(f.Container, "cali40all-hosts-net")
+				}, "5s", "200ms").Should(Equal(3))
+			}
+
+			Eventually(func() error {
+				err := externalClient.ExecMayFail("ip", "tunnel", "add", "tunl0", "mode", "ipip")
+				if err != nil && strings.Contains(err.Error(), "SIOCADDTUNNEL: File exists") {
+					return nil
 				}
+				return err
+			}).Should(Succeed())
+
+			externalClient.Exec("ip", "link", "set", "tunl0", "up")
+			externalClient.Exec("ip", "addr", "add", "dev", "tunl0", "10.65.222.1")
+			externalClient.Exec("ip", "route", "add", "10.65.0.0/24", "via",
+				felixes[0].IP, "dev", "tunl0", "onlink")
+
+			felixes[0].Exec("ip", "route", "add", "10.65.222.1", "via",
+				externalClient.IP, "dev", "tunl0", "onlink")
+		})
+
+		JustAfterEach(func() {
+			if CurrentGinkgoTestDescription().Failed {
+				externalClient.Exec("ip", "r")
+				externalClient.Exec("ip", "l")
+				externalClient.Exec("ip", "a")
 			}
 
 		})
+		AfterEach(func() {
+			externalClient.Stop()
+		})
 
 		It("should have all-hosts-net ipset configured with the external hosts and workloads connect", func() {
-			f := felixes[0]
-			// Add the ip route via tunnel back on the Felix for which we nuked when we removed its BGP spec.
-			f.Exec("ip", "route", "add", w[1].IP, "via", felixes[1].IP, "dev", "tunl0", "onlink")
-			cc.ExpectSome(w[0], w[1])
+			cc.ExpectSome(externalClient, w[0])
```

Comment:
The key thing that we were testing was that, without an entry in the IP set, IPIP should be blocked.  So, suggest you add a couple of lines to this test that check that removing the IP from the config breaks connectivity.

Reply:
sure I can add that :+1: